packages/block-library/src/navigation-submenu/edit.js

```diff
@@ -47,10 +47,26 @@ import { speak } from '@wordpress/a11y';
  */
 import { ItemSubmenuIcon } from './icons';
 import { name } from './block.json';
+import navigationLinkMetadata from '../navigation-link/block.json';
 
 const ALLOWED_BLOCKS = [ 'core/navigation-link', 'core/navigation-submenu' ];
 
-const DEFAULT_BLOCK = [ 'core/navigation-link' ];
+// Set default block to use default or undefined attribute values when inserted.
+const DEFAULT_BLOCK = [
+	'core/navigation-link',
+	{
+		...Object.entries( navigationLinkMetadata.attributes ).reduce(
+			( acc, attribute ) => ( {
+				...acc,
+				[ attribute[ 0 ] ]:
+					attribute[ 1 ].default !== undefined
+						? attribute[ 1 ].default
+						: undefined,
+			} ),
+			{}
+		),
+	},
```

Comment:
It's not great that we have to add exceptions to copied attributes in every single place we use direct insert. We may not be using it in a lot of core blocks (and there probably won't be many more), but if/when it becomes stable, we want third party block developers to find it easy to use.

One alternative would be to specify in DEFAULT_BLOCK a list of attributes we want copied, and have `getAdjacentBlockAttributes` only return those; then by default if nothing is specified nothing is copied. 

Or we could add an `__experimentalCopyStyleAttributes` instead of leveraging direct insert, but what we're doing here is close enough to direct insert that it kind of makes sense to have it there.

Reply:
> It's not great that we have to add exceptions to copied attributes in every single place we use direct insert.

Definitely not, thanks for taking a look here!

> One alternative would be to specify in DEFAULT_BLOCK a list of attributes we want copied, and have getAdjacentBlockAttributes only return those; then by default if nothing is specified nothing is copied.

Oh, I like that idea — I felt a bit uneasy just copying everything, and I think it makes it cleaner if we can keep it a part of `DEFAULT_BLOCK` if possible without adding an extra flag.

At the moment `DEFAULT_BLOCK` is an array so that it matches the arguments we pass to `createBlock`. I was wondering if it'd make sense to switch it to an object, so that we could add an extra key for the attributes to be copied, and have something like the following?

```
const DEFAULT_BLOCK = {
  name: 'core/my-block',
  attributes: {
    myDefaultAttribute: 'default value to set'
  },
  attributesToCopy: {
    myCopiedAttribute: true
  }
}
```

I'll keep chipping away at this PR next week, but let me know if you have a particular preference for how we implement it. Either way, explicitly flagging when we'd like to copy attributes sounds safer than assuming and having block authors have to add exceptions 🙂